src/components/TableCell.js

```diff
@@ -1,11 +1,13 @@
 export default {
     functional: true,
 
-    props: ['column', 'row'],
+    props: ['column', 'row', 'index'],
 
     render(createElement, { props }) {
         const data = {};
 
+        props.row.data['index'] = (props.index + 1);
```

Comment:
Not sure the `+ 1` is necessary. This can be solved with a formatter if the user needs 1-based indexing. A 0-based index meshes better with javascript array indexing.

Reply:
I would also call the data property `$index` instead of `index` since it's very probable that the user already has an index in his data, possibly called 'index'.